### ISSUE TITLE
Question regarding the behavior of the JsonConvert.DeserializeObject API

### ISSUE THREAD

**Issue by ElvisChen1986** [JamesNK/Newtonsoft.Json#2706]
Hallo, I have a question regarding the behavior of the JsonConvert.DeserializeObjectAPI.
If the instance of a list property would be created and assigned with some value in the default constructor, redundant items would be found after the deserialize is done.
e.g. in the case of the following code snippet, the value of the human.Hands.Count would be 4 after the deserialize. In my opinion, it should be 2 because there are 2 items in the json string. The deserialize behavior should override the existing one. I also tested other JSON API, e.g. System.Text.Json, they could return 2 after the deserialize is done.

Please let me know whether I misunderstood this API, thanks a lot.

namespace JsonTest4
{
    public class Hand
    {

    }

    public class Human
    {
        public List<Hand> Hands{ get; set; }

        public Human()
        {
            Hands = new List<Hand>()
            {
                new Hand(),
                new Hand()
            };
        }
    }

    class Program
    {
        const string ss= "{\"Hands\":[{\"Name\":\"Left\"},{\"Name\":\"Right\"}]}";

        static void Main(string[] args)
        {
            Human human = JsonConvert.DeserializeObject(ss, typeof(Human)) as Human;

            Console.WriteLine(human.Hands.Count);
            Console.ReadKey(false);
        }
    }
}

**Comment by elgonzo** [comment]
Yeah, it's an example of Newtonsoft.Json's documentation not making a good job of making new users of the library aware of the default (de)serialization settings and their behavior, which can sometimes be different from what many would intuitively expect them to be.

If you look at the documentation for (de)serialization setting [`ObjectCreationHandling`](https://www.newtonsoft.com/json/help/html/SerializationSettings.htm#ObjectCreationHandling), you'll notice that its default value is _Auto_, which behaves as follows (excerpt from the documentation):
> **add** JSON values to existing collections during deserialization. 

So, in case of your List\<T\> example, it should now be clear how you end up with four elements: The two initial elements in the list plus the two elements from the json **added** during deserialization.

Also another bit the documentation does not really clarify is that if the target collection is something like an array (basically a collection which does not allow adding of elements) then Newtonsoft.Json will try to replace the existing "no-add" collection with a new collection instance containing the deserialized collection elements (permitting that the respective field/property can actually being set, or a parameterized constructor is being used during deserialization for passing the new collection instance). 

Which all basically means, the observed behavior can change depending on the concrete collection type at play here as well as whether it would be possible for the deserializer to pass a new collection instance into the target field/property/constructor in case the collection type does not support adding of elements.
